packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/update.test.ts

```diff
@@ -0,0 +1,741 @@
+            require_alias: true,
+            body: expect.objectContaining({
+              multiNamespaceIsolatedType: { title: 'Testing' },
+              namespaces: ['default'],
+              references: [],
+              type: 'multiNamespaceIsolatedType',
+              ...mockTimestampFields,
+            }),
+          }),
+          expect.anything()
+        );
+      });
+
+      it(`doesn't accept custom references if not an array`, async () => {
+        const test = async (references: unknown) => {
+          migrator.migrateDocument.mockImplementation(mockMigrateDocumentForUpdate);
+          await updateSuccess(client, repository, registry, type, id, attributes, {
+            // @ts-expect-error references is unknown
+            references,
+          });
+          expect(
+            (client.index.mock.calls[0][0] as estypes.CreateRequest<SavedObjectsRawDocSource>).body!
+              .references
+          ).toEqual([]);
+          client.index.mockClear();
+          client.create.mockClear();
+        };
+        await test('string');
+        await test(123);
+        await test(true);
+        await test(null);
+      });
+
+      it(`defaults to a refresh setting of wait_for`, async () => {
+        migrator.migrateDocument.mockImplementation(mockMigrateDocumentForUpdate);
+        await updateSuccess(client, repository, registry, type, id, { foo: 'bar' });
+        expect(client.index).toHaveBeenCalledWith(
+          expect.objectContaining({
+            refresh: 'wait_for',
+          }),
+          expect.anything()
+        );
+      });
+
+      it(`defaults to the version of the existing document when type is multi-namespace`, async () => {
+        migrator.migrateDocument.mockImplementation(mockMigrateDocumentForUpdate);
+        await updateSuccess(
+          client,
+          repository,
+          registry,
+          MULTI_NAMESPACE_ISOLATED_TYPE,
+          id,
+          attributes,
+          { references }
+        );
+        const versionProperties = {
+          if_seq_no: mockVersionProps._seq_no,
+          if_primary_term: mockVersionProps._primary_term,
+        };
+        expect(client.index).toHaveBeenCalledWith(
+          expect.objectContaining(versionProperties),
+          expect.anything()
+        );
+      });
+
+      it(`accepts version`, async () => {
+        migrator.migrateDocument.mockImplementation(mockMigrateDocumentForUpdate);
+        await updateSuccess(client, repository, registry, type, id, attributes, {
+          version: encodeHitVersion({ _seq_no: 100, _primary_term: 200 }),
+        });
+        expect(client.index).toHaveBeenCalledWith(
+          expect.objectContaining({ if_seq_no: 100, if_primary_term: 200 }),
+          expect.anything()
+        );
+      });
+
+      it('retries the operation in case of conflict error', async () => {
+        client.get.mockResponse(getMockGetResponse(registry, { type, id }));
+
+        client.index
```

Comment:
Added unit tests for the retry-on-conflict behavior, given it would be fairly challenging to add tests for this at an higher level.